File: system-test/sec-tests/python/sec_10_ldap_injection.py
# SEC-10: LDAP Injection — filter concatenation
# Test: TC-CR-016
import ldap

def search_user(username):
    conn = ldap.initialize('ldap://localhost')
    search_base = "dc=example,dc=com"
    query = "(uid=" + username + ")"
    return conn.search_s(search_base, ldap.SCOPE_SUBTREE, query)

File: system-test/sec-tests/typescript/sec_10_ldap_injection.ts
// SEC-10: LDAP Injection
// Test: TC-CR-016
import { createClient } from 'ldapjs';

function searchUser(username: string) {
  const client = createClient({ url: 'ldap://localhost:389' });
  const opts = {
    filter: `(uid=${username})`,  // Injection
    scope: 'sub' as const
  };
  client.search('dc=example,dc=com', opts, (err, res) => {});
}
